test/qpy_compat/run_tests.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+set -e
+set -x
+
+python -m venv qiskit_venv
+qiskit_venv/bin/pip install ../..
+
+for version in $(git tag --sort=-creatordate) ; do
+    parts=( ${version//./ } )
+    if [[ ${parts[1]} < 18 ]] ; then
```

Comment:
```suggestion
    if [[ ${parts[1]} -lt 18 ]] ; then
```

Correcting a bigger problem, actually - `<` is a string-comparison operator, so `[[ 2 < 18 ]]` is false-y.

Reply:
Because, you know, having the _mathematical symbol_ `<` be the string comparison operator and the _string_ `-lt` be the mathematical comparison was the sensible choice.

Reply:
Heh, I always get them backwards. One day I'll get it right :)